subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -41,6 +59,95 @@ static struct bt_uuid_128 smp_bt_svc_uuid = BT_UUID_INIT_128(
 static struct bt_uuid_128 smp_bt_chr_uuid = BT_UUID_INIT_128(
 	BT_UUID_128_ENCODE(0xda2e7828, 0xfbce, 0x4e01, 0xae9e, 0x261174997c48));
 
+/* Helper function that allocates conn_param_data for a conn. */
+static struct conn_param_data *alloc_conn_param_data(struct bt_conn *conn)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(conn_data); i++) {
+		if (conn_data[i].conn == NULL) {
+			conn_data[i].conn = conn;
+			return &conn_data[i];
+		}
+	}
+
+	/* Conn data must exists. */
+	__ASSERT_NO_MSG(false);
+	return NULL;
+}
+
+/* Helper function that returns conn_param_data associated with a conn. */
+static struct conn_param_data *get_conn_param_data(const struct bt_conn *conn)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(conn_data); i++) {
+		if (conn_data[i].conn == conn) {
+			return &conn_data[i];
+		}
+	}
+
+	/* Conn data must exists. */
+	__ASSERT_NO_MSG(false);
+	return NULL;
+}
+
+/* Sets connection parameters for a given conn. */
+static void set_conn_latency(struct bt_conn *conn, bool low_latency)
+{
+	struct bt_le_conn_param params;
+	struct conn_param_data *cpd;
+	struct bt_conn_info info;
+	int ret = 0;
+
+	cpd = get_conn_param_data(conn);
+
+	ret = bt_conn_get_info(conn, &info);
+	__ASSERT_NO_MSG(!ret);
+
+	if ((low_latency && (info.le.latency == 0)) ||
+	    ((!low_latency) && (info.le.latency == DEFAULT_LATENCY))) {
+		/* Already updated. */
+		return;
+	}
+
+	params.interval_min = info.le.interval;
+	params.interval_max = info.le.interval;
+	params.latency = (low_latency) ? (0) : (DEFAULT_LATENCY);
```

Comment:
I'm a bit confused by this, the speed of mcumgr is determined by the connection intervals, not the latency. The latency is the number of packets that the peripheral device can skip transmitting if it has no data to send, so I'm not understanding how this improves the mcumgr transfer speed over BLE

Reply:
The peripheral has no data to send, and if the central is not indicating more data to send through the MD bit then peripheral  will use the peripheral latency and skip connection events.

This is more of a workaround for the MCUMGR_SMP not being efficient at transmitting.